test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs

```diff
@@ -336,6 +338,90 @@ public HttpConnectionDispatcherTests(ITestOutputHelper output) : base(output)
+                    context.TraceIdentifier = "requestid";
+                    context.Connection.Id = "connectionid";
+                    context.Connection.LocalIpAddress = IPAddress.Loopback;
+                    context.Connection.LocalPort = 4563;
+                    context.Connection.RemoteIpAddress = IPAddress.IPv6Any;
+                    context.Connection.RemotePort = 43456;
+
+                    var builder = new ConnectionBuilder(services.BuildServiceProvider());
+                    builder.UseEndPoint<HttpContextEndPoint>();
+                    var app = builder.Build();
+
+                    // Start a poll
+                    var task = dispatcher.ExecuteAsync(context, new HttpSocketOptions(), app);
+
+                    // Send to the application
+                    var buffer = Encoding.UTF8.GetBytes("Hello World");
+                    await connection.Application.Output.WriteAsync(buffer);
+
+                    // The poll request should end
+                    await task;
+
+                    // Now do a new send again without the poll (that request should have ended)
+                    await connection.Application.Output.WriteAsync(buffer);
+
+                    connection.Application.Output.Complete();
+
+                    // Wait for the endpoint to end
+                    await connection.ApplicationTask;
+
+                    var connectionHttpContext = connection.GetHttpContext();
+                    Assert.NotNull(connectionHttpContext);
+
+                    Assert.Equal(2, connectionHttpContext.Request.Query.Count);
+                    Assert.Equal(connection.ConnectionId, connectionHttpContext.Request.Query["id"]);
+                    Assert.Equal("value", connectionHttpContext.Request.Query["another"]);
+
+                    Assert.Equal(3, connectionHttpContext.Request.Headers.Count);
+                    Assert.Equal("h1", connectionHttpContext.Request.Headers["header1"]);
+                    Assert.Equal("h2", connectionHttpContext.Request.Headers["header2"]);
+                    Assert.Equal("h3", connectionHttpContext.Request.Headers["header3"]);
+                    Assert.Equal("requestid", connectionHttpContext.TraceIdentifier);
+                    Assert.Equal("claimValue", connectionHttpContext.User.Claims.FirstOrDefault().Value);
+                    Assert.Equal("connectionid", connectionHttpContext.Connection.Id);
+                    Assert.Equal(IPAddress.Loopback, connectionHttpContext.Connection.LocalIpAddress);
+                    Assert.Equal(4563, connectionHttpContext.Connection.LocalPort);
+                    Assert.Equal(IPAddress.IPv6Any, connectionHttpContext.Connection.RemoteIpAddress);
+                    Assert.Equal(43456, connectionHttpContext.Connection.RemotePort);
+                    Assert.NotNull(connectionHttpContext.RequestServices);
```

Comment:
Also check that doing stuff with the response has no effect?

Reply:
I can assert the properties are null.